hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperator.java

```diff
@@ -329,25 +326,10 @@ private void throwModificationInReadOnlyException() {
     private void publishWanReplicationEvent() {
         assert entryWasModified();
 
-        Data dataKey = toHeapData(this.dataKey);
-
         if (eventType == REMOVED) {
-            if (backup) {
-                mapEventPublisher.publishWanReplicationRemoveBackup(mapName, dataKey, Clock.currentTimeMillis());
-            } else {
-                mapEventPublisher.publishWanReplicationRemove(mapName, dataKey, Clock.currentTimeMillis());
-            }
-
-            return;
-        }
-
-        Record record = recordStore.getRecord(dataKey);
-        Data dataNewValue = toHeapData(ss.toData(newValue));
-        EntryView entryView = createSimpleEntryView(dataKey, dataNewValue, record);
-        if (backup) {
-            mapEventPublisher.publishWanReplicationUpdateBackup(mapName, entryView);
+            mapOperation.publishWanRemove(dataKey);
```

Comment:
The original call here ended up in a NOP backup method implementation. This call fires remove events at the end of the call chain.

Reply:
handled inside `MapEventPublisherImpl#publishWanUpdate` method.